Effects/DynamicShadows/source/PlayState.hx

```diff
@@ -0,0 +1,261 @@
+package;
+
+import flixel.addons.nape.FlxNapeSpace;
+import flixel.addons.nape.FlxNapeTilemap;
+import flixel.FlxG;
+import flixel.FlxSprite;
+import flixel.FlxState;
+import flixel.math.FlxPoint;
+import flixel.math.FlxRandom;
+import flixel.text.FlxText;
+import flixel.tile.FlxTilemap;
+import flixel.util.FlxColor;
+import nape.geom.Vec2;
+import nape.geom.Vec2List;
+import openfl.display.BlendMode;
+import openfl.geom.ColorTransform;
+import openfl.geom.Matrix;
+
+using flixel.util.FlxSpriteUtil;
+
+/*
+ * This was based on a guide from this forum post: http://forums.tigsource.com/index.php?topic=8803.0
+ * Ported to HaxeFlixel by Xerosugar
+ * 
+ * If you're feeling up the challenge, here's how YOU help can improve this demo:
+ * - Make it possible to extends the shadows to the edge of the screen
+ * - Make it possible to use multiple light sources while still maintaining a decent frame rate
+ * - Improve the performance of *this* demo
+ * - Make it possible to blur the edges of the shadows
+ * - Make it possible to limit the light source's influence, or "strength"
+ * - Your own ideas? :)
+ * 
+ * @author Tommy Elfving
+ */
+
+class PlayState extends FlxState
+{
+	public static inline var TILE_SIZE:Int = 16;
+	private static inline var PROP_BARREL:Int = 5;
+	private static inline var PROP_GEM:Int = 6;
+	
+	/**
+	 * Only contains non-collidabe tiles
+	 */
+	private var background:FlxTilemap;
+	
+	/**
+	 * The layer into which the actual "level" will be drawn, and also the one objects will collide with
+	 */
+	private var foreground:FlxNapeTilemap;
+	
+	/**
+	 * The sprite that shadows will be drawn to
+	 */
+	private var shadowCanvas:FlxSprite;
+	
+	/**
+	 * The sprite that the actual darkness and the gem's flare-like effect will be drawn to
+	 */
+	private var shadowOverlay:FlxSprite;
+	
+	/**
+	 * The light source!
+	 */
+	private var gem:Gem;
+	
+	/**
+	 * Used later when randomly moving the red flares that the gem emits
+	 */
+	private var random:FlxRandom;
+	
+	private var shadowColor = 0xff2a2963;
+	private var overlayColor = 0xff887fff;
+	
+	/**
+	 * If there's a small gap between something (could be two tiles, even if they're right next to each other), this should cover it up for us
+	 */
+	private var lineStyle:LineStyle;
+	
+	/**
+	 * Used for drawing stuff
+	 */
+	private var drawStyle:DrawStyle;
+	
+	private var infoText:FlxText;
+	
+	override public function create():Void
+	{
+		super.create();
+		
+		random = new FlxRandom();
+		FlxG.camera.bgColor = 0x5a81ad;
+		lineStyle = { color:shadowColor, thickness:1 };
+		drawStyle = {matrix:new Matrix(), colorTransform:new ColorTransform(1,1,1)};
+		
+		FlxNapeSpace.init();
+		FlxNapeSpace.space.gravity.setxy(0, 1200);
+		FlxNapeSpace.drawDebug = false; // You can toggle this on/off one by pressing 'D'
```

Comment:
That's the default already - though this just made me realize that it isn't explicitly being initialized which could cause issues on neko. :)